packages/core/src/SearchInput.js

```diff
@@ -18,100 +18,148 @@ export const BEM = {
-        }
-    }
+    isFoucs = false;
+
+    isControlled = () => (typeof this.props.value) !== 'undefined';
 
     handleInputChange = (event) => {
-        this.setState({ inputValue: event.target.value });
+        const { searchWhenInputChange, onChange, onSearch } = this.props;
+        if (this.isControlled()) {
+            onChange(event);
+        } else {
+            this.setState({ innerValue: event.target.value });
+        }
+
+        if (searchWhenInputChange) {
+            onSearch(event.target.value);
+        }
     }
 
     handleResetButtonClick = () => {
-        this.setState({ inputValue: '' }, () => this.notifySearch());
+        this.inputRef.current.focus();
+
+        const { onReset, value } = this.props;
+        const { innerValue } = this.state;
+
+        if (this.isControlled()) {
+            onReset(value);
+        } else {
+            onReset(innerValue);
+            this.setState({ innerValue: '' });
+        }
+    }
+
+    handleSearch = () => {
+        const { onSearch, value } = this.props;
+        const { innerValue } = this.state;
+
+        onSearch(this.isControlled() ? value : innerValue);
+    }
+
+    handleInputFocus = () => {
+        this.isFoucs = true;
     }
 
     handleInputBlur = () => {
-        this.notifySearch();
+        this.isFoucs = false;
+        const { searchWhenInputBlur } = this.props;
+        if (searchWhenInputBlur) {
+            // Prevent triggering `onSearch` when reset button clicked.
+            setTimeout(() => {
+                if (!this.isFoucs) {
+                    this.handleSearch();
+                }
+            }, 30);
+        }
     }
 
     handleInputKeyup = (event) => {
         if (event.key === 'Enter') {
-            this.notifySearch();
+            this.handleSearch();
         }
     }
 
-    renderResetButton() {
-        return (
-            <button
-                type="button"
-                className={`${BEM.resetBtn}`}
-                aria-label="Reset"
-                tabIndex="-1"
-                onClick={this.handleResetButtonClick}>
-                <Icon type="delete" color="gray" />
-            </button>
-        );
-    }
-
     render() {
-        const { input: inputProps, placeholder, className } = this.props;
-        const { inputValue } = this.state;
+        const { inputProps, value, placeholder, className } = this.props;
+        const { innerValue } = this.state;
 
-        const rootClassName = classNames(className, `${BEM.root}`);
+        const inputValue = this.isControlled() ? value : innerValue;
         const isLoading = this.context.status === STATUS_CODE.LOADING;
+        const rootClassName = classNames(className, `${BEM.root}`);
 
         return (
             <div className={rootClassName}>
                 <div className={BEM.inputWrapper}>
-                    <Icon type="search" />
+                    <Icon type="search" onClick={this.handleSearch} className={`${BEM.icon}`} />
```

Comment:
It doesn't seem to be a common UI practice though, please consider if user can know it's clickable?

Reply:
Yeah, I removed it.